cdap-client/src/main/java/co/cask/cdap/client/AuthorizationClient.java

```diff
@@ -198,10 +198,13 @@ private void executeExistingRolesRequest(Role role, HttpRequest request) throws
     }
   }
 
-  private HttpResponse executePrivilegeRequest(EntityId entityId, HttpRequest request) throws FeatureDisabledException,
-    UnauthenticatedException, IOException, NotFoundException, UnauthorizedException {
+  private HttpResponse executePrivilegeRequest(EntityId entityId, Principal principal, HttpRequest request) throws
+    FeatureDisabledException, UnauthenticatedException, IOException, NotFoundException, UnauthorizedException {
     HttpResponse httpResponse = doExecuteRequest(request, HttpURLConnection.HTTP_NOT_FOUND);
     if (HttpURLConnection.HTTP_NOT_FOUND == httpResponse.getResponseCode()) {
+      if (!listAllRoles().contains(new Role(principal.getName()))) {
+        throw new NotFoundException("Role: " + principal.getName());
```

Comment:
there is a `RoleNotFoundException`.
@rohitsinha54  @bdmogal 
Can we simply make that extend NotFoundException and use that here instead?